packages/typespec-ts/src/modular/emitSamples.ts

```diff
@@ -0,0 +1,515 @@
+import { NoTarget } from "@typespec/compiler";
+import { isArm } from "../utils/clientUtils.js";
+import {
+  buildPropertyNameMapper,
+  isSpreadBodyParameter
+} from "./helpers/typeHelpers.js";
+import { getClassicalClientName } from "./helpers/namingHelpers.js";
+import {
+  hasKeyCredential,
+  hasTokenCredential
+} from "../utils/credentialUtils.js";
+
+/**
+ * Interfaces for samples generations
+ */
+interface ExampleValue {
+  name: string;
+  value: string;
+  isOptional: boolean;
+}
+
+interface EmitSampleOptions {
+  topLevelClient: SdkClientType<SdkServiceOperation>;
+  generatedFiles: SourceFile[];
+  classicalMethodPrefix?: string;
+}
+/**
+ * Helpers to emit samples
+ */
+export function emitSamples(dpgContext: SdkContext): SourceFile[] {
+  const generatedFiles: SourceFile[] = [];
+  for (const client of dpgContext.sdkPackage.clients) {
+    emitClientSamplesRecursively(dpgContext, client, {
+      topLevelClient: client,
+      generatedFiles
+    });
+  }
+  return generatedFiles;
+}
+
+function emitClientSamplesRecursively(
+  dpgContext: SdkContext,
+  client: SdkClientType<SdkServiceOperation>,
+  options: EmitSampleOptions
+) {
+  for (const operationOrGroup of client.methods) {
+    // handle client-level methods
+    if (operationOrGroup.kind !== "clientaccessor") {
+      emitMethodSamples(dpgContext, operationOrGroup, options);
+      continue;
+    }
+    // handle operation group
+    let prefix = normalizeName(
+      operationOrGroup.response.name,
+      NameType.Property
+    );
+    // append hierarchy prefix if hierarchyClient is enabled
+    if (dpgContext.rlcOptions?.hierarchyClient === true) {
+      prefix =
+        (options.classicalMethodPrefix
+          ? `${options.classicalMethodPrefix}.`
+          : "") + prefix;
+    }
+
+    emitClientSamplesRecursively(dpgContext, operationOrGroup.response, {
+      ...options,
+      classicalMethodPrefix: prefix
+    });
+  }
+}
+
+function emitMethodSamples(
+  dpgContext: SdkContext,
+  method: SdkServiceMethod<SdkServiceOperation>,
+  options: EmitSampleOptions
+): SourceFile | undefined {
+  const examples = method.operation.examples ?? [];
+  if (examples.length === 0) {
+    return;
+  }
+  const project = useContext("outputProject");
+  const operationPrefix = `${options.classicalMethodPrefix ?? ""} ${
+    method.name
+  }`;
+  const sampleFolder = join(
+    dpgContext.generationPathDetail?.rootDir ?? "",
+    "samples-dev"
+  );
+  const fileName = normalizeName(`${operationPrefix} Sample`, NameType.File);
+  const sourceFile = project.createSourceFile(
+    join(sampleFolder, `${fileName}.ts`),
+    "",
+    {
+      overwrite: true
+    }
+  );
+  const exampleFunctions = [];
+  // TODO: remove hard-coded for package
+  if (dpgContext.rlcOptions?.packageDetails?.name) {
+    sourceFile.addImportDeclaration({
+      moduleSpecifier: dpgContext.rlcOptions?.packageDetails?.name,
+      namedImports: [getClassicalClientName(options.topLevelClient)]
```

Comment:
@joheredi Any advice on how to resolve this dependency? This would be taken as an external dependency for samples which reply on our generated client. But the name `NetworkAnalyticsClient ` would only be calculated in runtime and the package name is configed in tspconfig.

```ts
import { NetworkAnalyticsClient } from "@azure/arm-networkanalytics";
```

Reply:
[Issue ](https://github.com/Azure/autorest.typescript/issues/2821)created to track this.